workflow/controller/artifact_gc.go

```diff
@@ -451,6 +451,7 @@ func (woc *wfOperationCtx) createArtifactGCPod(ctx context.Context, strategy wfv
 						RunAsUser:                pointer.Int64(8737),
 						ReadOnlyRootFilesystem:   pointer.Bool(true),
 						AllowPrivilegeEscalation: pointer.Bool(false),
+						SeccompProfile:           &corev1.SeccompProfile{Type: "RuntimeDefault"},
```

Comment:
Is `seccomp` supported everywhere? AppArmor & SELinux aren't IIRC. Want to make sure we don't break compatibility

Also if we set this here, we should really set this on the `agent` and on `resource` containers as well (all 3 are locked down -- perhaps some of those could be refactored into a common config)

Reply:
Thank you for the hint regarding the `podSpecPatch` that did fix our initial problem. 

<details>
  <summary>For everyone finding this in the future, these are the changes I made to our workflows that produce artifacts</summary>
  
```
apiVersion: argoproj.io/v1alpha1
kind: CronWorkflow
metadata:
  name: some-workflow
  namespace: some-namespace
spec:
  workflowSpec:
    artifactGC:
      podSpecPatch: |
        securityContext:
          seccompProfile:
            type: RuntimeDefault
```
</details>

Setting this for every workflow might not be the prettiest solutions but it works for us. 

After a short little search  it seems like seccomp was first introduced in the linux kernel in version 2.6.12 in 2005 and should be present in every more or less recent linux distro.
But i cant be 100% certain and i dont know how k8s would handle this scenario.

How do you want to continue with this?
If  you want to continue with this, i would implement the changes you suggested above.

Reply:
It seems like it should be supported in every k8s distro afaict. It seems like if a CRI doesn't support seccomp the setting will just be ignored.
 And yes all non-EoL Linux support it.

I can't find a possible compat issue from a handful of searches either, so it seems safe enough to include, particularly as a feature.
Based on that, let's go ahead and implement it on this and the other two containers too. We'll get feedback on the RCs if there are compat issues.

Do you also want to file a follow-up issue to create more specific seccomp profiles for each container? Since we should be able to determine all their possible syscalls ahead of time. And then for newer k8s versions could [distribute those profiles via annotations](https://kubernetes.io/blog/2024/03/07/cri-o-seccomp-oci-artifacts/)

Reply:
> Setting this for every workflow might not be the prettiest solutions but it works for us.

You can use [`workflowDefaults`](https://argo-workflows.readthedocs.io/en/latest/default-workflow-specs/) to set this controller-wide

Reply:
Sorry, it took me a while to come back to this.

I refactored the SecurityContext for Pod and Container into a common function
(my thought to a function was because auf immutability)

Added this in the agent and the artifact_gc.

I couldn't find the `resource` you talked about, can you give me a hint on where to look?

Reply:
> that makes sense to me off-hand, but that's the one place I'll need to double-check if there's a better place to put it / name it / etc

small naming adjustment, but otherwise I think this is a good place to put it. we should probably split `common.go` into more files for it to be easier to find things. nice choice 🙂

Reply:
> Do you also want to file a follow-up issue to create more specific seccomp profiles for each container? Since we should be able to determine all their possible syscalls ahead of time. And then for newer k8s versions could [distribute those profiles via annotations](https://kubernetes.io/blog/2024/03/07/cri-o-seccomp-oci-artifacts/)


fyi: created the new issue: https://github.com/argoproj/argo-workflows/issues/13045